ChessGame/static/chessgame/js/ChessBoard/chessBoard.js

```diff
@@ -7,6 +7,8 @@ class ChessBoard {
   whiteCaptured = document.createElement('div');
   blackCaptured = document.createElement('div');
```

Comment:
I am having the hardest time every checking for opponents being in check and preventing pieces moving in check - I might want to restructure my data due to this

Reply:
Oooh this is a really interesting problem. I'd keep another 2D array of bools that holds the state of the board with respect to the King and whether it's in check or not. Definitely not an easy problem to solve.

Reply:
You might try looking at open source implementations of Chess as well and see how they do it. Good practice on reading / understanding other people's code

Reply:
Or maybe I'll just store the king's position